src/gen_rules.ml

```diff
@@ -1143,10 +1161,30 @@ let gen ~contexts ?(filter_out_optional_stanzas_with_missing_deps=true)
         ~stanzas
     in
     let module M = Gen(struct let sctx = sctx end) in
-    (Super_context.rules sctx, (context.name, stanzas)))
+    (context.name, ((module M : Gen), stanzas)))
   |> Future.all
   >>| fun l ->
-  let rules, context_names_and_stanzas = List.split l in
+  let map = String_map.of_alist_exn l in
+  let generators = String_map.map map ~f:fst in
   Alias.Store.unlink aliases unlink_aliases;
-  (Alias.rules aliases @ List.concat rules,
-   String_map.of_alist_exn context_names_and_stanzas)
+  let context_names = List.map contexts ~f:(fun c -> c.Context.name) in
+  let merlin_context = List.find contexts ~f:(fun c -> c.merlin) in
+  let gen_rules _build_system ~dir =
```

Comment:
Are we going to need this _build_system parameter elsewhere? It doesn't seem like it's used anywhere currently.

Reply:
Nope. But given that `Build_system.create` takes the `gen_rules` callback as argument, it seems to me that the callback should get the build system as argument. It's not used in jbuilder, but eventually this stuff might be exposed as a library.